### PR TITLE
Add a method inv(CuMatrix)

### PR DESCRIPTION
**Pull request by amontoison**

### CODE REVIEW

File: lib/cusolver/linalg.jl [hunk 1]
```diff
@@ -342,3 +342,14 @@ end
...
 LinearAlgebra.cholcopy(A::LinearAlgebra.RealHermSymComplexHerm{<:Any,<:CuArray}) =
     copyto!(similar(A, LinearAlgebra.choltype(A)), A)
+
+## inv
+
+function LinearAlgebra.inv(A::StridedCuMatrix{T}) where T <: BlasFloat
+    n = checksquare(A)
+    F = copy(A)
+    factors, ipiv, info = getrf!(F)
+    B = CuMatrix{T}(I, n, n)
```
Comment (amontoison) [inline lib/cusolver/linalg.jl:352]: @maleadt Do you know a better way to create an identity matrix of size `n` with CUDA.jl?

**Comment by amontoison** [comment]
CUSOLVER also has an optimized routine to directly invert triangular matrices (trtri). I started to interface it [here](https://github.com/JuliaGPU/CUDA.jl/pull/2074).

File: lib/cusolver/linalg.jl (hunk 1 above)
Comment (maleadt) [reply lib/cusolver/linalg.jl:352]: This looks fine, and is specialized in GPUArrays. We could avoid a kernel to set the diagonal by using a 2D memcpy, but looking at the profiler most of the time is spent during the zero initialization, so there's not much to gain.

**Comment by maleadt** [comment]
Thanks! This has been often requested, so is great to have.

**Pull request merged**